### PR TITLE
WIP local development docs

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

**Comment by rynowak** [comment]
Sharing my progress so you can comment on it :)

**Comment by rynowak** [comment]
There's a rough TOC at the bottom of the additional points to discuss. Please point it out if I missed anything.

**Comment by davidfowl** [comment]
This is a really great start and I think we should merge it and tbd the rest. The only feedback I have is that I wonder if docs like this get too long. Maybe we need to break it up but I don’t have suggestions on how yet

File: docs/reference/local_development.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+- Load the application (`tye.yaml`)
+- Build all projects 
+- Pull all container images
+- Assign ports and bindings
+- Launch and monitor all services
+
+## Loading the application
+
+A Tye application is specified using `tye.yaml`. You can find more information about the Tye application format [here](/docs/reference/schema.md).
```
Comment (jkotalik) [inline docs/reference/local_development.md:21]: Let's figure out a way to share this info with tye deploy. I would end up duplicating most of this.

File: docs/reference/local_development.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+Tye acts as a *local orchestrator* for development by building and launching multiple services, managing and monitoring services, and enabling network communication between services.
+
+This document will describe how these features work, as well as the implications and behaviors of various optional settings.
+
+## Executing `tye run`
+
+When executing `tye run` for local development, Tye goes through the following steps:
+
+- Load the application (`tye.yaml`)
+- Build all projects
```
Comment (jkotalik) [inline docs/reference/local_development.md:14]: Should we mention extensions here?

Comment (rynowak) [reply docs/reference/local_development.md:14]: I wasn't planning to yet. That seems like a level of detail we don't need.

**Pull request merged**